### PR TITLE
Convert CSV file for elements doc to JSON (#460)

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses #460.

The Python code to do the job is the following:
```
import json
import numpy
import pandas as pd
import sys

for in_name in sys.argv[1:]:
  if in_name.endswith(".csv"):
    out_name = in_name[:len(in_name)-4]
  else:
    out_name = in_name
    in_name += ".csv"
  out_name += ".json"
  df = pd.read_csv(in_name, encoding="utf-8")
  inherits = []
  default_prop = None
  properties = []
  for row in list(df.to_records(index=False)):
    prop_name = row[0]
    if prop_name[0] == "*":
      if default_prop:
        raise ValueError(f"Multiple default property in {in_name}")
      prop_name = prop_name[1:]
      default_prop = prop_name
    elif prop_name[0] == ">":
      inherits.append(prop_name[1:])
      prop_name = None
    if prop_name:
      # This weird == test is to detect NaNs
      default_value = row[2] if row[2] == row[2] else None
      if type(default_value) is numpy.bool_:
        default_value = bool(default_value)
      properties.append({
        "name": prop_name,
        "type": row[1],
        "default": default_value,
        "doc": row[3]
      })
  element = {
    "default": default_prop
  }
  if inherits:
    element["inherits"] = inherits
  if properties:
    element["properties"] = properties
  with open(out_name, "w") as output_file:
    output_file.write(json.dumps(element, indent=2))
    print(f"{out_name} written.")
```

### CODE REVIEW

**Review by FredLL-Avaiga — APPROVED** [review]
I thought we'd have a unique json fille

**Comment by FabienLelaquais** [comment]
> 

That would make inheritance tricky to code, wouldn't it?
If you think that Taipy Studio would benefit from a unique file, that would be pretty easy (and simplify the doc generation process...).
Please let me know. I can certainly change that.

File: gui/doc/viselements.json [hunk 1]
```diff
@@ -0,0 +1,1252 @@
+{
+  "controls": [
+    [
+      "text",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json:4]: the construct [[name, description]] does not garantee the unicity. What's the rational behind that choice ?

File: gui/doc/viselements.json [hunk 2]
```diff
@@ -0,0 +1,1252 @@
...
+      {
+        "default": "value",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: Why not mark the property as default: true ?

File: gui/doc/viselements.json [hunk 3]
```diff
@@ -0,0 +1,1252 @@
...
+            "doc": "Column name for the <i>base</i> value. Used in bar charts only.\nSee the Plotly <a href=\"https://plotly.com/javascript/reference/bar/#bar-base\">bar chart base</a> documentation for details.\""
+          },
+          {
+            "name": "title",
+            "type": "str",
+            "doc": "The title of this chart control."
+          },
+          {
+            "name": "render",
+            "type": "bool",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: should be dynamic(bool)

File: gui/doc/viselements.json [hunk 4]
```diff
@@ -0,0 +1,1252 @@
...
+      {
+        "default": "render",
+        "inherits": [
+          "partial",
+          "shared"
+        ],
+        "properties": [
+          {
+            "name": "render",
+            "type": "bool",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: dynamic(bool)

File: gui/doc/viselements.json [hunk 5]
```diff
@@ -0,0 +1,1252 @@
...
+      {
+        "default": "value",
+        "inherits": [
+          "partial",
+          "shared"
+        ],
+        "properties": [
+          {
+            "name": "value",
+            "type": "str",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: dynamic

File: gui/doc/viselements.json [hunk 6]
```diff
@@ -0,0 +1,1252 @@
...
+            "doc": "Title of this block element."
+          },
+          {
+            "name": "expanded",
+            "type": "bool",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: dynamic

File: gui/doc/viselements.json [hunk 7]
```diff
@@ -0,0 +1,1252 @@
...
+        "inherits": [
+          "partial",
+          "on_change",
+          "active",
+          "shared"
+        ],
+        "properties": [
+          {
+            "name": "open",
+            "type": "bool",
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: dynamic

File: gui/doc/viselements.json [hunk 8]
```diff
@@ -0,0 +1,1252 @@
...
+            "name": "persistent",
+            "type": "bool",
+            "default": false,
+            "doc": "If True, the pane appears next to the page.\nIf False, the pane covers the page where it appeared."
+          }
+        ]
+      }
+    ]
+  ],
+  "others": [
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json]: naming ?

File: gui/doc/viselements.json [hunk 9]
```diff
@@ -0,0 +1,1252 @@
...
+  "controls": [
+    [
+      "text",
+      {
+        "default": "value",
+        "inherits": [
+          "on_change",
+          "shared"
+        ],
+        "properties": [
```
Comment (FredLL-Avaiga) [inline gui/doc/viselements.json:10]: properties: { name: {... details}} would insure unicity

**Review by FredLL-Avaiga — APPROVED** [review]
Is the structure fixed by the doc requirements ?

File: gui/doc/viselements.json (hunk 9 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json:10]: But I need to keep the order...

Comment (FredLL-Avaiga) [reply gui/doc/viselements.json:10]: I see : order or unicity, you chose
Another choice would be to add an index property but it's a pain ...

**Comment by FabienLelaquais** [comment]
> 

No but the information has to be somewhere.
With this new structure, the order of elements controls the order they appear in the doc.

File: gui/doc/viselements.json (hunk 2 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: I like that!

File: gui/doc/viselements.json (hunk 3 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: 👍

File: gui/doc/viselements.json (hunk 4 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: 👍

File: gui/doc/viselements.json (hunk 5 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: 👍

File: gui/doc/viselements.json (hunk 6 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: 👍

File: gui/doc/viselements.json (hunk 9 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json:10]: Correct. Another requirement is that it should be quite easy to update...

File: gui/doc/viselements.json (hunk 8 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: -> "undocumented"

File: gui/doc/viselements.json (hunk 7 above)
Comment (FabienLelaquais) [reply gui/doc/viselements.json]: 👍

**Review by FredLL-Avaiga — APPROVED** [review]
No order needed for properties then ?

**Comment by FabienLelaquais** [comment]
> No order needed for properties then ?

Yes yes! Order is required in the viselements.json file: this is used when generating the properties section for each element.

**Pull request merged**